torchgen/gen.py

```diff
@@ -1330,6 +1331,60 @@ def maybe_create_view_group(
     return list(concatMap(maybe_create_view_group, grouped_by_views.values()))
 
 
+def get_composite_graph(
+    grouped_native_functions: Sequence[Union[NativeFunction, NativeFunctionsGroup]]
+) -> CompositeGraph:
+    def nativefunction_with_group(
+        g: Union[NativeFunction, NativeFunctionsGroup]
+    ) -> List[Tuple[NativeFunction, Optional[NativeFunctionsGroup]]]:
+        if isinstance(g, NativeFunction):
+            fs = [g]
+            group = None
+        elif isinstance(g, NativeFunctionsGroup):
+            fs = list(g.functions())
+            group = g
+        else:
+            assert_never(g)
+        return [(f, group) for f in fs]
+
+    graph: CompositeGraph = {}
+    nativefunction_for = {
+        f.func.name: (f, group)
+        for g in grouped_native_functions
+        for f, group in nativefunction_with_group(g)
+    }
+
+    for g in grouped_native_functions:
+        for f, _ in nativefunction_with_group(g):
+            if len(f.composite) == 0:
```

Comment:
Should this use `should_gen_dispatchless_composite()`?

Reply:
Not in this context, since it depends on whether a kernel is registered or not to a dispatch key.
That said, I'm implementing the changes Ed suggested, i.e. moving dispatch-less code to its own thing. This should move it to a file in the `dest` folder, and build the graph only for kernels that should be generated (as you suggest, here).